gcc/d/gdc.texi

```diff
@@ -120,196 +120,662 @@ and the Info entries for @file{gdc} and @file{gcc}.
+Enforces the @code{@@property} syntax in D.
+
+@item -fsyntax-only
+@cindex @option{-fsyntax-only}
+@cindex @option{-fno-syntax-only}
+Check the code for syntax errors, but do not actually compile it.  This
+only suppresses the generation of the object code, and can be used in
+conjunction with @option{-fdoc} or @option{-fintfc} options.
+
+@item -ftransition=@var{id}
+@cindex @option{-ftransition}
+Report additional information about D language changes identified by
+@var{id}.  The following values are supported:
+
+@table @samp
+@item all
+@cindex @option{-ftransition=all}
+List information on all language changes.
+@item checkimports
+@cindex @option{-ftransition=checkimports}
+Give deprecation messages about @option{-ftransition=import} anomalies.
+@item complex
+@cindex @option{-ftransition=complex}
+List all usages of complex or imaginary types.
+@item dip1000
+@cindex @option{-ftransition=dip1000}
+Implements @uref{http://wiki.dlang.org/DIP1000} (experimental).
+@item dip25
+@cindex @option{-ftransition=dip25}
+Implements @uref{http://wiki.dlang.org/DIP25} (experimental).
+@item field
+@cindex @option{-ftransition=field}
+List all non-mutable fields which occupy an object instance.
+@item import
+@cindex @option{-ftransition=import}
+Revert to single phase name lookup.
+@item nogc
+@cindex @option{-ftransition=nogc}
+List all hidden GC allocations.
+@item tls
+@cindex @option{-ftransition=tls}
+List all variables going into thread local storage.
+@end table
+
+@end table
+
+@node Linking
+@section Options for Linking
+@cindex options, linking
+@cindex linking, static
+
+These options come into play when the compiler links object files into an
+executable output file. They are meaningless if the compiler is not doing
+a link step.
+
+@table @gcctabopt
+
+@item -defaultlib @var{libname}
+@cindex @option{-defaultlib}
+Specify the library to use instead of libphobos when linking.  Options
+specifying the linkage of libphobos, such as @option{-static-libphobos}
+or @option{-shared-libphobos}, are ignored.
+
+@item -debuglib
+@cindex @option{-debuglib}
+Specify the debug library to use instead of libphobos when linking.
+This option has no effect unless the @option{-g} option was also given
+on the command line.  Options specifying the linkage of libphobos, such
+as @option{-static-libphobos} or @option{-shared-libphobos}, are ignored.
+
+@item -nophoboslib
+@cindex @option{-nophoboslib}
+Do not use the Phobos standard runtime library when linking.  Only the
+libraries you specify are passed to the linker.  Options specifying the
+linkage of libphobos, such as @option{-static-libphobos}
+or @option{-shared-libphobos}, are ignored.
+
+@item -shared-libphobos
+@cindex @option{-shared-libphobos}
+On systems that provide @file{libphobos} as a shared and a static
+library, this option forces the use of the shared version. If no
+shared version of @file{libphobos} was built when the compiler was
+configured, this option has no effect.
```

Comment:
I'd be more specific here as this may be important for distribution package maintainers.  When the compiler is configured libgphobos doesn't even exist yet.

Instead I'd mention that these options only have an effect if supported by the linker and both shared and static versions are installed (if only one is installed, behaviour will be linker dependent).